runtime/src/qio/qio_formatted.c

```diff
@@ -361,6 +361,88 @@ qioerr _append_char(char* restrict * restrict buf, size_t* restrict buf_len, siz
   return 0;
 }
 
+qioerr read_entire_file(const int threadsafe,qio_channel_t* restrict ch, const char* restrict* restrict out, int64_t* restrict len_out, ssize_t maxlen)
+{  
+  qioerr err,err_reading_file=0;
+  int64_t file_length;
+  char* ret=NULL;
+  ssize_t amt=0;
+  err_t errcode;
+  int64_t fileLenGuess=1024;
+  ssize_t total_amt_read=0;
+  if( threadsafe ) 
+  {
+    err = qio_lock(&ch->lock);
+    if( err ) return err;
+  }
+
+//  err = qio_channel_mark(false, ch);
+//  if( err ) goto unlock;
+
+  err=qio_file_length(ch->file,&file_length);
+  //printf("file_length=%zd\n",file_length);
+  //err=qio_int_to_err(1);
+  if(!err)
+  {
+    //printf("no error\n");
+    ret=qio_malloc(file_length+1);
+    ret[0]='\0';
+    err=qio_channel_read(false, ch, ret,file_length+1, &amt);
+    if(qio_err_to_int(err)==EEOF)
+    {
+       err=0;
+    }
+    else
+    {
+       err=qio_int_to_err(1);
+    }
+    total_amt_read=amt;
+  }
+
+  if(err)
+  {    
+    ret=qio_realloc(ret,total_amt_read+fileLenGuess+1);
+    while(true)
+    {
+      err_reading_file = qio_channel_read(false, ch,&ret[total_amt_read],1024LL, &amt);
+      
+      //printf("amt=%zd\n",amt);
+      total_amt_read+=amt;
+      if (err_reading_file == QIO_EEOF) break;
+      fileLenGuess=total_amt_read+1024;
+      ret = qio_realloc(ret, fileLenGuess+1);
+    }
+    file_length=total_amt_read;
+    ret=qio_realloc(ret,file_length+1);
+    //err= qio_channel_read(false, ch, ret, file_length, &amt);
+  }
+    
+  //printf("file_length reassigned=%zd\n",file_length);
+  ret[file_length] = '\0'; // always add terminator at the end
+    
+  err= 0;
+  unlock:
```

Comment:
let's remove the `unlock:` since it appears unused

Reply:
removed